lib/foreman_maintain/package_manager/yum.rb

```diff
@@ -0,0 +1,129 @@
+module ForemanMaintain::PackageManager
+  class Yum < Base
+    VERSIONLOCK_START_CLAUSE = '## foreman-maintain - start'.freeze
+    VERSIONLOCK_END_CLAUSE = '## foreman-maintain - end'.freeze
+    VERSIONLOCK_CONFIG_FILE = '/etc/yum/pluginconf.d/versionlock.conf'.freeze
+    VERSIONLOCK_DEFAULT_LIST_FILE = '/etc/yum/pluginconf.d/versionlock.list'.freeze
+
+    def foreman_related_packages
+      installed = sys.execute("repoquery -a --qf='%{envra} %{repo.baseurl}'").
+                  split("\n"). map(&:split)
+      repos_to_lock = []
+      foreman_repo = installed.find { |package| package[0] =~ /foreman-installer-\d+/ }[1]
```

Comment:
I feel this find iteration can be avoided by simple grep command ? 
# repoquery -a --qf='%{envra} %{repo.baseurl}'|grep foreman-installer

Reply:
This method is trying to find the repo foreman-installer is comming from. Then it selects for locking all installed packages coming from that repo.
This part could be simplified as you described but then I'd need to run another repoquery to get the list of packages from the detected repo. The repoquery is pretty expensive call so it is more effective to run it once and process the result twice in ruby.

Reply:
I did some more experiments and optimized repoquery so that we can save 10-15% of time and also a few megs of memory with the suggested approach. But that is only in case we are locking packages from single repo. For more then one repo it is 20+% slower. So I'd wait till we figure out if we need to lock also scl/tools/other. If we won't I'll rewrite using two repoqueries. Is it acceptable @upadhyeammit.

Reply:
@mbacovsky this approach is acceptable.